groups/script.js

```diff
@@ -1,436 +1,248 @@
+          dataStore.filteredGroupsIds = Object.values(dataStore.groups).map(
+            (group) => group.id,
+          );
+        } else {
+          const search = value.toLowerCase();
+          dataStore.filteredGroupsIds = Object.values(dataStore.groups)
+            .filter((group) => group.title.toLowerCase().includes(search))
+            .map((group) => group.id);
+        }
+        obj[prop] = value;
+        break;
+      case 'isGroupCreationModalOpen':
+        obj[prop] = value;
+
+        if (value) {
+          renderGroupCreationModal({
+            onClose: () => (dataStore.isGroupCreationModalOpen = false),
+            onSubmit: (inputValue) => {
+              return createDiscordGroupRole({
+                rolename: inputValue,
+              }).then(() => {
+                showToaster('Group created successfully');
+                dataStore.isGroupCreationModalOpen = false;
+
+                // Future improvement: Use a more robust way to refresh the data
+                setTimeout(() => location.reload(), 2500);
+              });
+            },
+          });
+        } else {
+          removeGroupCreationModal();
+        }
+        break;
+      case 'userSelf':
+        obj[prop] = value;
+        break;
+      case 'discordId':
+        obj[prop] = value;
+        break;
+      default:
+        throw new Error('Invalid property set');
+    }
+    return true;
+  },
+};
 
-/**
- * Create DOM for "created by author" line under groupName
- *
- */
-const createAuthorDetailsDOM = (firstName, lastName, imageUrl) => {
-  const container = document.createElement('div');
-  container.classList.add('created-by--container');
+const dataStore = new Proxy(
+  {
+    userSelf: null,
+    groups: null,
+    filteredGroupsIds: null,
+    search: '',
+    discordId: null,
+    isCreateGroupModalOpen: false,
+  },
+  handler,
+);
 
-  if (imageUrl) {
-    const userAvatar = document.createElement('img');
-    userAvatar.classList.add('created-by--avatar');
-    userAvatar.src = imageUrl;
-    userAvatar.setAttribute('alt', "group's creator image");
-    container.appendChild(userAvatar);
-  }
+// Lifecycle functions
 
-  if (firstName || lastName) {
-    const createdBy = document.createElement('span');
-    createdBy.classList.add('created-by');
-    createdBy.textContent = `created by ${firstName ?? ''} ${lastName ?? ''}`;
-    container.appendChild(createdBy);
-  }
+const onCreate = () => {
+  renderLoadingCards();
+  renderLoadingNavbarProfile();
 
-  return container;
+  getUserSelf()
+    .then((data) => {
+      if (data.statusCode === 401) {
+        renderNavbarProfileSignin();
+        renderNotAuthenticatedPage();
+        return;
+      } else if (data.error) {
+        throw new Error(data);
+      }
+      dataStore.userSelf = data;
+      afterAuthentication();
+    })
+    .catch((err) => {
+      console.error(err);
+      removeLoadingCards();
+    })
+    .finally(() => removeLoadingNavbarProfile());
```

Comment:
It's a good practice to catch specific errors and handle them accordingly rather than catching all errors with a generic catch block.

Reply:
Got it. How do you think should we handle the error here?